### PR TITLE
refactor certificates to use ResourceForm

### PR DESCRIPTION
**Pull request by qbalukom**

### CODE REVIEW

File: core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js [hunk 1]
```diff
@@ -1,63 +1,189 @@
...
+              >
+                {csrIsEncoded
+                  ? t('secrets.buttons.decode')
+                  : t('secrets.buttons.encode')}
+              </Button>
+            )}
+          </div>
+        )}
+      />
+      {withCSR && (
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js:85]: The CSR doesn't work properly - If I type something into the `CSR` input, then change the switch, the previously typed-in value is still in the yaml

File: core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js [hunk 2]
```diff
@@ -1,63 +1,189 @@
...
+          tooltipContent={t('certificates.tooltips.secret-name')}
+          propertyPath="$.spec.secretRef.name"
+          input={Inputs.Text}
+          inputProps={{
+            placeholder: t('certificates.placeholders.secret-name'),
+          }}
+        />
+      )}
+      {existingSecret && (
+        <SecretRef
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js:179]: The search doesn't work as expected, I cannot see all namespaces there

Comment (dariadomagala) [reply core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js:179]: Also, shouldn't we only allow current namespace then?
Screen Shot 2021-10-04 at 14 42 57

**Review by dariadomagala — COMMENTED** [review]
Additionally:
Screen Shot 2021-10-04 at 13 31 22

File: core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js [hunk 3]
```diff
@@ -1,63 +1,189 @@
...
+      />
+      {withCSR && (
+        <>
+          <ResourceForm.FormField
+            required
+            label={t('certificates.csr')}
+            tooltipContent={t('certificates.tooltips.csr')}
+            propertyPath="$.spec.csr"
+            input={({ value, setValue }) => {
+              console.log('CSR', value);
```
[the remark this answers is not in the document]
Comment (qbalukom) [reply core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js:93]: oops

File: core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js (hunk 2 above)
Comment (qbalukom) [reply core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js:179]: i think this one might be related to specific implementation, not the api itself...

File: core-ui/src/shared/components/ResourceRef/ExternalResourceRef.js [hunk 4]
```diff
@@ -54,68 +54,52 @@ export function ExternalResourceRef({
...
-        </div>
-        <div className="fd-col fd-col-md--7">
-          <Dropdown
-            id="external-resource-dropdown"
+      <ResourceForm.FormField
+        required={required}
+        label={t('common.labels.resource-namespace', { resource: labelPrefix })}
+        tooltipContent={t('common.tooltips.secret-ref-namespace')}
+        input={() => (
+          <ComboboxInput
```
Comment (dariadomagala) [inline core-ui/src/shared/components/ResourceRef/ExternalResourceRef.js:66]: maybe we could use also the `selectionType='auto-inline' `prop?

File: core-ui/src/shared/components/ResourceRef/ExternalResourceRef.js [hunk 5]
```diff
@@ -54,68 +54,52 @@ export function ExternalResourceRef({
...
-          <Dropdown
-            id="external-resource-dropdown"
+        )}
+      />
+      <ResourceForm.FormField
+        required={required}
+        label={t('common.labels.resource-name', { resource: labelPrefix })}
+        tooltipContent={t('common.tooltips.secret-ref-name')}
+        input={() => (
+          <ComboboxInput
```
Comment (dariadomagala) [inline core-ui/src/shared/components/ResourceRef/ExternalResourceRef.js:87]: If I choose the secret name first, and the namespace is not set, I have to choose the namespace manually. So, I can set a namespace that doesn't have this secret. Maybe we could either detect and choose a namespace based on a chosen secret or disable the secret dropdown until we choose a namespace?

File: core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js [hunk 6]
```diff
@@ -1,63 +1,236 @@
...
+      jp.value(certificate, '$.spec.secretRef.name', undefined);
+    } else {
+      jp.value(certificate, '$.spec.secretRef', undefined);
+    }
+    setCertificate(certificate);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [existingSecret]);
+
+  useEffect(() => {
+    setWithCSR(!!jp.value(certificate, '$.spec.csr'));
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js:62]: If I start typing the `CSR` value and then delete the whole string, the CSR toggle is set to false.

**Review by dariadomagala — CHANGES_REQUESTED** [review]

**Review by dariadomagala — COMMENTED** [review]
Screen Shot 2021-10-07 at 07 51 06
please fix the red alert csses

File: core/src/i18n/en.yaml [hunk 7]
```diff
@@ -168,13 +169,25 @@ certificates:
...
   placeholders:
     common-name: Certificate CN (max 64 characters)
     csr: Plain text Certificate Signing Request
     dns-names: Additional domain names, one per line
     encoded-csr: Base64-encoded Certificate Signing Request
     name: Certificate name
     secret-name: Requested Secret name
+  tooltips:
+    common-name: Common Name is the primary domain name of the certificate.
+    csr: Certificate signing request is an alternative way to provide Common Name, DNSNames and other information.
```
Comment (grego952) [inline core/src/i18n/en.yaml:183]: DNS Names (space missing)

File: core/src/i18n/en.yaml [hunk 8]
```diff
@@ -168,13 +169,25 @@ certificates:
...
+    with-csr: Enable to use a Certificate Signing Request instead of providing domain names manually.
+    dns-names: DNS Names are the optional additional domain names of the certificate.
+    renew: Trigger a certificate renewal if enabled.
+    issuer: Certificate Issuer. If no issuer is given, default issuer will be used.
+    existing-secret: Use an exising secret for storing the certificate if enabled. Otherwise a new secret will be created.
```
Comment (grego952) [inline core/src/i18n/en.yaml:188]: Secret

File: core/src/i18n/en.yaml [hunk 9]
```diff
@@ -168,13 +169,25 @@ certificates:
...
+    secret-name: Name of the secret object to use for storing the certificate.
```
Comment (grego952) [inline core/src/i18n/en.yaml:189]: Secret

File: core/src/i18n/en.yaml [hunk 10]
```diff
@@ -319,6 +334,8 @@ common:
     no-instances-found-namespace: No matching {{value}} found in this Namespace.
     parse-error: Parse error, previous valid input will be used.
     type-to-select: Start typing to select {{value}} from the list.
+    resource-namespace-error: Requested namespace not found.
```
Comment (grego952) [inline core/src/i18n/en.yaml:337]: Namespace

File: core/src/i18n/en.yaml [hunk 11]
```diff
@@ -319,6 +334,8 @@ common:
...
+    resource-name-error: Requested resource with this name not found in the selected namespace.
```
Comment (grego952) [inline core/src/i18n/en.yaml:338]: Namespace

File: core/src/i18n/en.yaml [hunk 12]
```diff
@@ -168,13 +169,25 @@ certificates:
     messages:
       failure: Failed to create the certificate.
```
Comment (grego952) [inline core/src/i18n/en.yaml:170]: Certificate

File: core/src/i18n/en.yaml (hunk 7 above)
Comment (grego952) [inline core/src/i18n/en.yaml:183]:
```suggestion
    csr: Certificate Signing Request is an alternative way to provide Common Name, DNS Names, and other information.
```

**Review by dariadomagala — DISMISSED** [review]

File: core/src/i18n/en.yaml [hunk 13]
```diff
@@ -166,15 +167,27 @@ certificates:
...
   placeholders:
     common-name: Certificate CN (max 64 characters)
     csr: Plain text Certificate Signing Request
     dns-names: Additional domain names, one per line
     encoded-csr: Base64-encoded Certificate Signing Request
     name: Certificate name
     secret-name: Requested Secret name
+  tooltips:
+    common-name: Common Name is the primary domain name of the Certificate.
+    csr: Certificate Signing Request is an alternative way to provide Common Name, DNS Names, and other information.
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:183]:
```suggestion
    csr: Certificate Signing Request is an alternative way to provide common name, DNS names, and other information.
```

why capitalized?

File: core/src/i18n/en.yaml [hunk 14]
```diff
@@ -166,15 +167,27 @@ certificates:
...
+    with-csr: Enable to use a Certificate Signing Request instead of providing domain names manually.
+    dns-names: DNS Names are the optional additional domain names of the Certificate.
+    renew: Trigger a certificate renewal if enabled.
+    issuer: Certificate Issuer. If no issuer is given, default issuer will be used.
+    existing-secret: Use an existing Secret for storing the certificate if enabled. Otherwise a new Secret will be created.
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:188]: Yep, Secret is indeed a word that we [capitalize](https://kyma-project.io/community/guidelines/content-guidelines/08-style-and-terminology#capitalized-terms)! :)

**Review by Wawrzyn321 — APPROVED** [review]

**Pull request merged**